src/test/java/TestTransactionsCalculatorService.java

```diff
@@ -0,0 +1,87 @@
+import static org.junit.Assert.assertEquals;
+
+import fruitshop.dao.FruitShopStorageDaoImpl;
+import fruitshop.db.FruitShopStorage;
+import fruitshop.model.FruitReport;
+import fruitshop.model.FruitTransaction;
+import fruitshop.model.Operation;
+import fruitshop.service.impl.TransactionsCalculatorServiceImpl;
+import fruitshop.strategy.operation.handlers.impl.BalanceOperation;
+import fruitshop.strategy.operation.handlers.impl.PurchaseOperation;
+import fruitshop.strategy.operation.handlers.impl.ReturnOperation;
+import fruitshop.strategy.operation.handlers.impl.SupplyOperation;
+import fruitshop.strategy.operation.impl.OperationStrategyImpl;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestTransactionsCalculatorService {
+    private static final TransactionsCalculatorServiceImpl transactionsCalculator =
+            new TransactionsCalculatorServiceImpl(
+                    new OperationStrategyImpl(
+                            Map.of(
+                                    Operation.BALANCE, new BalanceOperation(),
+                                    Operation.PURCHASE, new PurchaseOperation(),
+                                    Operation.RETURN, new ReturnOperation(),
+                                    Operation.SUPPLY, new SupplyOperation()
+                            )),
+                    new FruitShopStorageDaoImpl());
+
+    @Before
+    public void setUp() {
+        FruitShopStorage.fruitTransactions.clear();
+    }
+
+    @Test
+    public void calculate_Basic_isOk() {
+        FruitShopStorage.fruitTransactions.addAll(List.of(
+                new FruitTransaction(Operation.getByValue('b'), "banana", 20),
+                new FruitTransaction(Operation.getByValue('b'), "apple", 100),
+                new FruitTransaction(Operation.getByValue('s'), "banana", 100),
+                new FruitTransaction(Operation.getByValue('p'), "banana", 13),
+                new FruitTransaction(Operation.getByValue('r'), "apple", 10),
+                new FruitTransaction(Operation.getByValue('p'), "apple", 20),
+                new FruitTransaction(Operation.getByValue('p'), "banana", 5),
+                new FruitTransaction(Operation.getByValue('s'), "banana", 50)
```

Comment:
we dont need many tests with the same logic

u can delete it / and delete in other casses